### PR TITLE
Fix broken function ProductStockSyncResource::post

### PR DESCRIPTION
**Pull request by malachy-mcconnell**
And stock sync should update one or many

### CODE REVIEW

File: modules/sku/src/ProductManager.php [hunk 1]
```diff
@@ -564,7 +576,7 @@ public function synchronizeStockData(array $stock = [], $storeId = '') {
       'success' => TRUE,
     ];
 
-    return (new ResourceResponse($response));
+    return ($response);
```
Comment (miromichalicka) [inline modules/sku/src/ProductManager.php:579]: please change this back, it needs to be ResourceResponse because of caching

Comment (malachy-mcconnell) [reply modules/sku/src/ProductManager.php:579]: I wondered about that. Here the `$response` is returned as an array into `\Drupal\acm_sku\Plugin\rest\resource\ProductSyncResource::post`where the response is formed and sent back.

There are two resources, yes? In `\Drupal\acm_sku\Plugin\rest\resource\ProductSyncResource::post` Should I use: (for non-cached I think)
1. `Drupal\rest\ModifiedResourceResponse;`
or (for cached, yes?)
2. `Drupal\rest\ResourceResponse;`

We should always use 2., right?

Comment (nikunjkotecha) [reply modules/sku/src/ProductManager.php:579]: @malachy-mcconnell we faced issues with ResourceResponse, for any reason if Drupal's render() is used, it kept shouting about leaked metadata detected error so we changed to ModifiedResourceResponse (used even in CORE) everywhere. For stock I don't think we will ever have metadata issue but to keep it consistent and to be sure there won't be such error we used ModifiedResourceResponse

Comment (nikunjkotecha) [reply modules/sku/src/ProductManager.php:579]: check https://github.com/acquia/commerce-manager/pull/55

File: modules/sku/src/ProductManager.php [hunk 2]
```diff
@@ -523,38 +518,55 @@ public function synchronizeStockData(array $stock = [], $storeId = '') {
...
-      if (isset($stock['is_in_stock']) && empty($stock['is_in_stock'])) {
-        $stock['quantity'] = 0;
+      if (!isset($stock['sku']) || !strlen($stock['sku'])) {
+        $this->logger->error('Invalid or empty product SKU.');
+        return ($response);
       }
 
-      $quantity = isset($stock['quantity']) ? $stock['quantity'] : 0;
+      // Acquire lock to ensure parallel processes are executed one by one.
+      // TODO @Malachy: Should we change the key and move outside loop?
```
Comment (csibar) [inline modules/sku/src/ProductManager.php:541]: Please resolve if needed and remove TODO comment.

Comment (malachy-mcconnell) [reply modules/sku/src/ProductManager.php:541]: I'd like to leave the TODO in place. I don't know the answer and this seems like a place where efficiencies can be made in the future so I want to flag it as such.

**Pull request merged**